src/qml/QFieldCloudPopup.qml

```diff
@@ -304,6 +304,15 @@ Popup {
           titleText: detailsText.startsWith('[QF/') ? qsTr('A server error has occured, please try again.') : qsTr('A network error has occured, please try again.')
           detailsText: ''
 
+          function humanizeError(errorString) {
```

Comment:
Let's move that into a C++ static function attached to QFieldCloudUtils (here https://github.com/opengisch/QField/blob/master/src/core/utils/qfieldcloudutils.h). Make sure the human-friendly error messages are translatable.

Also, let's not check for the presence of eg "Quota Error" and instead try to find the code instead (i.e. "code":"over_quota").

@suricactus , what would be the best way to handle this here? string search, or trying to extract the json part?